hikey960: Initial commit v2

vchong opened the pull request:
Replaces https://github.com/OP-TEE/optee_os/pull/1681.

Please ignore first commit which is https://github.com/OP-TEE/optee_os/pull/1683.

Thanks!

vchong commented:
Comments addressed.

jforissier commented on core/arch/arm/plat-hikey/platform_config.h:
```diff
@@ -32,19 +32,39 @@
+#define PL011_UART6_BASE	0xFFF32000
+#if defined(CFG_CONSOLE_UART) && (CFG_CONSOLE_UART == 5)
+#define CONSOLE_UART_BASE	PL011_UART5_BASE
+#elif !defined(CFG_CONSOLE_UART) || (CFG_CONSOLE_UART == 6)
```
This can (and should) be `#elif defined(CFG_CONSOLE_UART) && (CFG_CONSOLE_UART == 6)` since `CFG_CONSOLE_UART` now has a default value set in `conf.mk`. The problem with the current code is that it will silently ignore any value different from 5 and 6 and use 6 in any case. Please fix line 41 (hikey-hikey) also.

vchong replied:
Good catch, thanks! In that case, can we simplify this to just `#elif CFG_CONSOLE_UART == 6` and etc?

vchong replied:
The only thing with the simplification is that if `CFG_CONSOLE_UART` is not defined, then it will default to uart0 for hikey instead of printing an error.

jforissier replied:
Yeah, `!defined(CFG_CONSOLE_UART)` is useless if we make sure it is always defined obviously.

jforissier commented:
Thanks @vchong, just one more comment, once it is addressed:
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

Ah, yes, one more question, maybe for @etienne-lms. Since Hikey960 has both A53 and A73 cores, is it OK to have `-mcpu=cortex-a53` (set in file `cortex-armv8-0.mk`)?

vchong commented:
Rebased, squashed and tags applied. Ready for merge. Thanks!

@jforissier You're welcome! Please make sure you're ok with the `CONSOLE_UART_BASE` section one last time before merging. Thanks!

vchong commented:
Sorry, please hold. One of the commit message needs to be changed.

vchong commented:
Commit message fixed. Thanks!

vchong commented:
Thanks!
